docs/docs/0.6.0/documentation/networking.md

```diff
@@ -87,4 +87,101 @@ In order for connectivity to work among the `kuma-dp` instances, Kuma also assum
+
+The `CIDR` field sets the IP range of virtual IPs. The default `240.0.0.0/4` is reserved for future use IPv4 range and is guaranteed to be non-routable. We strongly recommend to not change this, unless it is really needed.
+
+### Operation 
+
+The basic operation of Kuma DNS includes a couple of main components: DNS server, VIPs allocator, cross-replica persistence.
+
+The DNS server listens on port `5653` and reponds for type `A` DNS requests and answers with `A` record, e.g. `<service>.mesh.	60	IN	A	240.0.0.100`. The default TTL is set to 60 seconds, to ensure client will synchronize with Kuma DNS and account for any changes happening meanwhile.
+
+Kuma DNS allocates the virtual IPs from the configured CIDR, by constantly scanning the services available in all Kuma meshes. When a service is removed its VIP is freed too and Kuma DNS will not respond for it with `A` DNS record.
+
+Kuma DNS is optimized to work in a multi-replica setup, where a number of Kuma control planes are running in parallel. It leverages the internal leader election so that VIP allocation happens only on the leader, and uses an internal storage resource to ensure all other replicas are in sync with the available service/VIP mappings allocated by the leader.
+
+### Usage
+
+Consuming a service handled by Kuma DNS from inside a Kubernetes container looks like this:
+```bash
+<kuma-enabled-pod>$ curl http://echo-server_echo-example_svc_1010.mesh:80
+<kuma-enabled-pod>$ curl http://echo-server_echo-example_svc_1010.mesh
+```
+
+Since the default VIP created listenere will default to port `80`, it can be omitted when using a standard HTTP client.
+
+Effectively the result of the Kuma DNS operation is creating a virtual IP listener by patching the dataplane resource before it gets translated to Envoy xDS configuration. However, by inspecting `curl localhost:9901/config_dump`, we can see sections similar to this one:
+
+```json
+    {
+     "name": "outbound:240.0.0.1:80",
+     "active_state": {
+      "version_info": "51adf4e6-287e-491a-9ae2-e6eeaec4e982",
+      "listener": {
+       "@type": "type.googleapis.com/envoy.api.v2.Listener",
+       "name": "outbound:240.0.0.1:80",
+       "address": {
+        "socket_address": {
+         "address": "240.0.0.1",
+         "port_value": 80
+        }
+       },
+       "filter_chains": [
+        {
+         "filters": [
+          {
+           "name": "envoy.tcp_proxy",
+           "typed_config": {
+            "@type": "type.googleapis.com/envoy.config.filter.network.tcp_proxy.v2.TcpProxy",
+            "stat_prefix": "echo-server_kuma-test_svc_80",
+            "cluster": "echo-server_kuma-test_svc_80"
+           }
+          }
+         ]
+        }
+       ],
+       "deprecated_v1": {
+        "bind_to_port": false
+       },
+       "traffic_direction": "OUTBOUND"
+      },
+      "last_updated": "2020-07-06T14:32:59.732Z"
+     }
+    },
+```
```

Comment:
I think we also need to explain that Kuma DNS does NOT return all instances. It's not service discovery.

Reply:
Adding a sentence to Operation section.